internal/app/subsystems/api/service/request.go

```diff
@@ -58,3 +63,34 @@ type RejectPromiseHeader struct {
 type RejectPromiseBody struct {
 	Value promise.Value `json:"value"`
 }
+
+// Subscription
+
+type SearchSubscriptionsParams struct {
+	// Q      string `form:"q" json:"q"`
+	PromiseId string `json:"promiseId"`
+	Limit     int    `form:"limit" json:"limit"`
+	// Cursor    string `form:"cursor" json:"cursor"`
+}
+
+type CreateSubscriptionHeader struct {
+	RequestId      string                  `header:"request-id"` // qq: just for grpc?
```

Comment:
This is an optional parameter for tracking requests cross-systems. If provided this field is used in corresponding logs, if not provided I believe we generate a uuid for each request.